packages/dev/core/src/Materials/greasedLinePluginMaterial.ts

```diff
@@ -0,0 +1,707 @@
+            resolutionLineWidth.x = this._options.resolution.x;
+            resolutionLineWidth.y = this._options.resolution.y;
+        } else {
+            resolutionLineWidth.x = this._engine.getRenderWidth();
+            resolutionLineWidth.y = this._engine.getRenderHeight();
+        }
+        resolutionLineWidth.z = this._options.width ? this._options.width : this._options.sizeAttenuation ? 1 : 0.1;
+        uniformBuffer.updateVector3("grl_resolution_lineWidth", resolutionLineWidth);
+
+        const dashOptions = TmpVectors.Vector4[0];
+        dashOptions.x = GreasedLinePluginMaterial._BooleanToNumber(this._options.useDash);
+        dashOptions.y = this._dashArray ?? 0;
+        dashOptions.z = this._options.dashOffset ?? 0;
+        dashOptions.w = this._options.dashRatio ?? 0.5;
+        uniformBuffer.updateVector4("grl_dashOptions", dashOptions);
+
+        const colorModeVisibilityColorsWidthUseColors = TmpVectors.Vector4[1];
+        colorModeVisibilityColorsWidthUseColors.x = this._options.colorMode ?? GreasedLineMeshColorMode.COLOR_MODE_SET;
+        colorModeVisibilityColorsWidthUseColors.y = this._options.visibility ?? 1;
+        colorModeVisibilityColorsWidthUseColors.z = this._colorsTexture ? this._colorsTexture.getSize().width : 0;
+        colorModeVisibilityColorsWidthUseColors.w = GreasedLinePluginMaterial._BooleanToNumber(this._options.useColors);
+        uniformBuffer.updateVector4("grl_colorMode_visibility_colorsWidth_useColors", colorModeVisibilityColorsWidthUseColors);
+
+        if (this._options.color) {
+            uniformBuffer.updateColor3("grl_singleColor", this._options.color ?? Color3.White());
+        }
+
+        uniformBuffer.setTexture("grl_colors", this._colorsTexture ?? GreasedLinePluginMaterial._EmptyColorsTexture);
+    }
+
+    /**
+     * Prepare the defines
+     * @param defines
+     * @param _scene
+     * @param _mesh
+     */
+    prepareDefines(defines: MaterialGreasedLineDefines, _scene: Scene, _mesh: AbstractMesh) {
+        const options = this._options;
+        defines.GREASED_LINE_HAS_COLOR = !!options.color;
+        defines.GREASED_LINE_SIZE_ATTENUATION = options.sizeAttenuation ?? false;
+        defines.GREASED_LINE_COLOR_DISTRIBUTION_TYPE_LINE = options.colorDistributionType === GreasedLineMeshColorDistributionType.COLOR_DISTRIBUTION_TYPE_LINE;
+    }
+
+    /**
+     * Get the class name
+     * @returns class name
+     */
+    getClassName() {
+        return GreasedLinePluginMaterial.GREASED_LINE_MATERIAL_NAME;
+    }
+
+    /**
+     * Get shader code
+     * @param shaderType vertex/fragment
+     * @returns shader code
+     */
+    getCustomCode(shaderType: string): Nullable<{ [pointName: string]: string }> {
+        if (shaderType === "vertex") {
+            return {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                CUSTOM_VERTEX_DEFINITIONS: `
+                    attribute vec4 grl_previousAndSide;

```

Comment:
due to the amount of extra attributes, I wonder if there should be a check somewhere to ensure we are not over capacity ?

Reply:
The best place would probably be in `Engine.createEffect`, since we're passing it the list of all attributes: it would simply be a matter of testing the array size against the limit.

However, apart from displaying a warning or error, I'm not sure what we can do(?)